spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/util/RequestContentDataExtractor.java

```diff
@@ -78,12 +79,16 @@
 					.collect(Collectors.toList());
 			List<String> listOfOnlyQueryParams = queryParamsGroupedByName.get(key);
 
-			if (listOfOnlyQueryParams != null
-					&& !listOfOnlyQueryParams.containsAll(listOfAllParams)) {
-				listOfAllParams.removeAll(listOfOnlyQueryParams);
-				for (String value : listOfAllParams) {
-					builder.add(key,
-							new HttpEntity<>(value, newHttpHeaders(request, key)));
+			if(listOfOnlyQueryParams != null) {
```

Comment:
Can you maybe comment in words what this is doing? Even before the change I'm not quite sure.

Reply:
I mean I can explain what the code is doing....

`request.getParameterMap` returns query params and form data.  If there is form data (`listOfOnlyQueryParams` does not contain `listOfAllParams`) then we remove the query params from `listofAllParams` and add the form params to the Map.  What I am not clear on is what that map ends up being used for...